src/Resources/config/services.yml

```diff
@@ -222,6 +222,21 @@ services:
         calls:
             - ["setContainer", ["@service_container"]]
 
+    contao.security.backend_user_provider:
+        class: Contao\CoreBundle\Security\User\ContaoBackendUserProvider
+        public: false
+        arguments:
+            - "@contao.framework"
+
+    contao.security.switch_user_button_generator:
+        class: Contao\CoreBundle\Security\User\SwitchUserButtonGenerator
```

Comment:
To me this is an event (callback) listener and should be in the according namespace.